bandit/bandit_test.go

```diff
@@ -14,11 +14,80 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestBanditDialer_chooseDialerForDomain(t *testing.T) {
+	baseDialer := newTcpConnDialer()
+	type fields struct {
+		dialers []Dialer
+	}
+	type args struct {
+		network string
+		addr    string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   Dialer
+		want1  int
+	}{
+		{
+			name: "should return the first dialer if there's only one dialer",
+			fields: fields{
+				dialers: []Dialer{baseDialer},
+			},
+			args: args{
+				network: "tcp",
+				addr:    "localhost:8080",
+			},
+			want:  baseDialer,
+			want1: 0,
+		},
+		{
+			name: "choose the non-failing dialer if there are multiple dialers",
+			fields: fields{
+				dialers: []Dialer{
+					newFailingTcpConnDialer(),
+					newFailingTcpConnDialer(),
+					newFailingTcpConnDialer(),
+					newFailingTcpConnDialer(),
+					baseDialer,
+					newFailingTcpConnDialer(),
+					newFailingTcpConnDialer(),
+				},
+			},
+			args: args{
+				network: "tcp",
+				addr:    "localhost:8080",
+			},
+			want:  baseDialer,
+			want1: 4,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opts := Options{
+				Dialers: tt.fields.dialers,
+			}
+			o, err := New(opts)
+			if err != nil {
+				t.Fatal(err)
+			}
+			got, got1 := o.chooseDialerForDomain(tt.args.network, tt.args.addr)
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("BanditDialer.chooseDialerForDomain() got = %v, want %v", got, tt.want)
+			}
+			if got1 != tt.want1 {
+				t.Errorf("BanditDialer.chooseDialerForDomain() got1 = %v, want %v", got1, tt.want1)
+			}
```

Comment:
Please don't commit this suggestion directly from github otherwise it'll break/panic since we are not importing `"github.com/stretchr/testify/assert"`  at this file yet
```suggestion
                        assert.Equal(t, tt.want1, got1, "BanditDialer.chooseDialerForDomain() got1 = %v, want %v", got1, tt.want1)
```